src/include/sof/probe/probe.h

```diff
@@ -0,0 +1,86 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+#ifndef __SOF_PROBE_PROBE_H__
+#define __SOF_PROBE_PROBE_H__
+
+#include <config.h>
+
+#if CONFIG_PROBE
+
+#include <ipc/probe.h>
+
+/*
+ * \brief Initialize probes subsystem
+ *
+ * param[in,optional] extraction_probe_dma - DMA associated with extraction
+ *		      In case extraction_probe_dma is NULL, extraction probes
+ *		      are unavailable.
+ */
+int probe_init(struct probe_dma *extraction_probe_dma);
+
+/*
+ * \brief Deinitialize probes subsystem.
+ *
+ * Detach extraction DMA if was enabled. Return -EINVAL in case some probes
+ * are still in use.
+ */
+int probe_deinit(void);
+
+/*
+ * \brief Setup injection DMAs for probes.
+ *
+ * param[in] count - number of DMAs configured during this call
+ * param[in] probe_dma - Array of size 'count' with configuration data for DMAs
+ */
+int probe_dma_set(uint32_t count, struct probe_dma *probe_dma);
```

Comment:
Again, this needs to be orthoginal to usage. i.e. I would do 
```
set -> add
get -> info
remove -> remove
```
This give us APIs/IPC for adding and removing (self explanitory) and also for info.

Reply:
done